src/org/parosproxy/paros/core/scanner/VariantAbstractQuery.java

```diff
@@ -177,9 +183,13 @@ private String setParameter(HttpMessage msg, NameValuePair originalPair, String
             if (isAppended && i < getParamList().size() - 1) {
                 sb.append(parser.getDefaultKeyValuePairSeparator());
             }
+            query = sb.toString();
+        }
+        if (query == null) {

```

Comment:
Could use the length of `sb` instead (does not require converting the `StringBuilder` to a `String` each time in the for loop).